upgrade_guide/migrating_from_astyanax/configuration/README.md

```diff
@@ -0,0 +1,242 @@
+# Configuration
+
+## How Configuring the Java driver works
+
+The two basic components in the Java driver are the `Cluster` and the `Session`.
+The `Cluster` is the object to create first, and on to which apply all
+global configuration options. Connecting to the `Cluster` creates a
+`Session`. Queries are executed through the `Session`.
+
+The `Cluster` object then is to be viewed as the equivalent of the `AstyanaxContext`
+object. ’Starting’ an `AstyanaxContext` object typically returns a `Keyspace`
+object, the `Keyspace` object is the equivalent of the *Java driver*’s `Session`.
+
+Configuring a `Cluster` works with the *Builder* pattern. The `Builder` takes all
+the configurations into account before building the `Cluster`.
+
+Following are some examples of the most important configurations that were 
+possible with *Astyanax* and how to translate them into *DataStax Java driver*
+configurations.
+
+## Connection pools
+
+Configuration of connection pools in *Astyanax* are made through the
+`ConnectionPoolConfigurationImpl`. This object gathers important configurations
+that the *Java driver* has categorized in multiple *Option* and *Policy* kinds.
+
+### Connections pools internals
+Everything concerning the internal pools of connections to the *Cassandra nodes*
+will be gathered in the Java driver in the `PoolingOptions` :
+
+*Astyanax*:
+
+```java
+ConnectionPoolConfigurationImpl cpool =
+       new ConnectionPoolConfigurationImpl("myConnectionPool")
+               .setInitConnsPerHost(2)
+               .setMaxConnsPerHost(3)
+```
+
+*Java driver*:
+
+```java
+PoolingOptions poolingOptions =
+       new PoolingOptions()
+           .setMaxRequestsPerConnection(HostDistance.LOCAL, 1024)
+           .setCoreConnectionsPerHost(HostDistance.LOCAL, 2)
+           .setMaxConnectionsPerHost(HostDistance.LOCAL, 3);
+```
+
+Note that the *Java driver* allows multiple simultaneous requests on one single
+connection, as it is based upon the *Native protocol*, an asynchronous binary
+protocol that can handle up to 32768 simultaneous requests.
+
+### Timeouts
+
+Timeouts concerning requests, or connections will be part of the `SocketOptions`.
+
+*Astyanax*:
+
+```java
+ConnectionPoolConfigurationImpl cpool =
+       new ConnectionPoolConfigurationImpl("myConnectionPool")
+               .setSocketTimeout(3000)
+               .setConnectTimeout(3000)
+```
+
+*Java Driver :*
+
+```java
+SocketOptions so =
```

Comment:
In the past we have advised users to not decrease `SocketOptions#readTimeoutMillis` and to depend on the cassandra side timeouts instead and in particular not to set readTimeoutMillis to a value less than the cassandra timeouts.   If the client timeout is less than the cassandra timeout, you don't give RetryPolicy an opportunity to kick in.